homeworks/22-nextjs-1/src/app/components/AvatarContainer/AvatarContainer.tsx

```diff
@@ -0,0 +1,56 @@
+import { FaPlus } from "react-icons/fa";
+import style from "./AvatarContainer.module.css";
+import AvatarTile from "../AvatarTile/AvatarTile";
+import { useState } from "react";
+import { addAvatar as fetchNewAvatar } from "@/app/services/avatars";
+import ButtonRefresh from "../ButtonRefresh/ButtonRefresh";
+import { Avatar } from "@/app/lib/definitions";
+
+export default function AvatarContainer({ avatars: initialAvatars = [] }) {
+  const [avatars, setAvatars] = useState<Avatar[]>(initialAvatars);
+
+  const addAvatar = async () => {
+    const newAvatar = await fetchNewAvatar();
+    setAvatars((prevAvatars) => [...prevAvatars, newAvatar[0]]);
+  };
+
+  const refreshAvatar = async (id: string) => {
+    const newAvatar = await fetchNewAvatar();
+    const updateAvatars = avatars.map((avatar) =>
+      avatar.id === id ? { ...avatar, url: newAvatar[0].url } : avatar
+    );
+    setAvatars(updateAvatars);
+  };
+
+  const refreshAllAvatars = async () => {
```

Comment:
You have endpoint with a limit parameter. It is not OK to send multiple requests, if you can send only 1.